Add tenant quota report and widget

lpichler opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1299963
https://trello.com/c/fq0qvvtp/98-rfe-reporting-unable-to-report-on-cf-tenant-quotas

I added tenant quota report to VM Sprawl -> Candidates -> Tenant Quotas Report is this location ok @gtanzillo  ?

I added TenantQuota's model methods for formatting values in human format  (2 GB, 2 Count..), because col_formatters in yaml are not usable here, because for determining TenantQuota value we need also column `TenantQuota#name` to determine type of quota and unit (for example `mem_allocated` have unit GB `cpu_allocated` have unit Count, ..). (it seems that col_formatters are basically only input-output functions). Please correct me If I am wrong.

Screenshot(only enforced tenant quotas are )
report:
![report](https://cloud.githubusercontent.com/assets/14937244/13392044/3d4a9b66-ded8-11e5-812d-59c149f5bd5d.png)

widget:
![widget](https://cloud.githubusercontent.com/assets/14937244/13392050/42a9fe1c-ded8-11e5-8f13-2885d589cabe.png)

@gtanzillo Can you review it, please ? Thanks!

lpichler commented:
@miq-bot remove-label wip

gtanzillo commented:
@lpichler You'll need to add TenantQuota to the list of "reportable" models so that users can create new reports based on tenant quotas. Also, it would probably be helpful to be able to report on the tenant that the quota belongs to in the context of a tenant quota report. Here's a diff of my local change to make that happen:
```diff
diff --git a/app/models/miq_expression.rb b/app/models/miq_expression.rb
index 4b7a8c3..bde89fd 100644
--- a/app/models/miq_expression.rb
+++ b/app/models/miq_expression.rb
@@ -52,6 +52,7 @@ class MiqExpression
     StoragePerformance
     ManageIQ::Providers::CloudManager::Template
     ManageIQ::Providers::InfraManager::Template
+    TenantQuota
     User
     VimPerformanceTrend
     Vm
@@ -144,6 +145,7 @@ class MiqExpression
     storage_adapters
     storage_files
     switches
+    tenants
     users
     vms
     volumes
diff --git a/app/models/tenant_quota.rb b/app/models/tenant_quota.rb
index 3e72294..a36f388 100644
--- a/app/models/tenant_quota.rb
+++ b/app/models/tenant_quota.rb
@@ -55,6 +55,8 @@ class TenantQuota < ApplicationRecord
   virtual_column :allocated_formatted, :type => :float
   virtual_column :available_formatted, :type => :float
 
+  include ReportableMixin
+
   before_validation(:on => :create) do
     self.unit = default_unit unless unit.present?
   end
```

gtanzillo commented on app/models/tenant_quota.rb:
```diff
@@ -49,10 +49,41 @@ class TenantQuota < ApplicationRecord
     self.unit = default_unit unless unit.present?
   end
 
+  def quota_value_formatted(quota_value)
```
In not sure that we need to add this method for formatting the quota values. Shouldn't the built-in formats that you get on the "Formatting" tab in the report editor do this?

gtanzillo commented:
@lpichler I added a question for Brad in the trello card asking about the menu location for the report.

lpichler commented:
@gtanzillo For now, I updated report to be able create to custom report based on this report.

and also added it to the menu but it seems that we need always 2 level menu with need to have in 
(https://github.com/ManageIQ/manageiq/blob/master/app/controllers/application_controller.rb#Lhttps://github.com/ManageIQ/manageiq/blob/master/app/controllers/application_controller.rb#L983
`r_group` variable is 'Tenants - Tenant Quotas' which comes from name of folder where located yaml manageiq/product/reports/780_Tenants - Tenant Quotas/001_Tenant Quotas.yaml  ).  

![screen shot 2016-03-02 at 12 50 17](https://cloud.githubusercontent.com/assets/14937244/13459787/4484b368-e076-11e5-9d43-0e614c6700b3.png)

lpichler commented:
@gtanzillo Now I am building report based on 'Tenant' table. 

I also add special if leg for TenantQuota columns in `html.rb`. So it means that user can select fields like TenantQuotas#used,... in custom report and these fields will be formatted according to TenantQuotas#name(must be included in column selection) as we needed and other format method will be ignored - just only for these TenantQuotas's fields.

Output for basic report is basically same like in first comment.
What do you think ?

If you are ok with it, I'll fix specs.
Thanks!

gtanzillo commented:
@lpichler This looks good. Please proceed with this approach. Thanks.

lpichler commented:
@miq-bot remove-label wip

lpichler commented:
@gtanzillo I finished changes and updated specs.

gtanzillo commented:
Looks good! :+1: thanks @lpichler !
